src/taxonomy/TaxonomyListPage.jsx

```diff
@@ -64,10 +69,75 @@ const TaxonomyListHeaderButtons = () => {
   );
 };
 
+const OrganizationFilterSelector = ({
+  isOrganizationListLoaded,
+  organizationListData,
+  selectedOrgFilter,
+  setSelectedOrgFilter,
+}) => {
+  const intl = useIntl();
+  const isOrgSelected = (value) => (value === selectedOrgFilter ? <Check /> : null);
+  const selectOptions = [
+    <MenuItem
+      key="all-orgs-taxonomies"
+      className="x-small"
+      iconAfter={() => isOrgSelected('All taxonomies')}
+      onClick={() => setSelectedOrgFilter('All taxonomies')}
+    >
+      { isOrgSelected('All taxonomies')
+        ? intl.formatMessage(messages.orgInputSelectDefaultValue)
+        : intl.formatMessage(messages.orgAllValue)}
+    </MenuItem>,
+    <MenuItem
+      key="unassigned-taxonomies"
+      className="x-small"
+      iconAfter={() => isOrgSelected('Unassigned')}
+      onClick={() => setSelectedOrgFilter('Unassigned')}
+    >
+      { intl.formatMessage(messages.orgUnassignedValue) }
+    </MenuItem>,
+  ];
+
+  if (isOrganizationListLoaded && organizationListData) {
+    organizationListData.map(org => (
+      selectOptions.push(
+        <MenuItem
+          key={`${org}-taxonomies`}
+          className="x-small"
+          iconAfter={() => isOrgSelected(org)}
+          onClick={() => setSelectedOrgFilter(org)}
+        >
+          {org}
+        </MenuItem>,
+      )
+    ));
+  }
```

Comment:
Alternatively, you can do this (roughly):
```suggestion
  const selectOptions = <>
    <MenuItem
      key="all-orgs-taxonomies"
      className="x-small"
      iconAfter={() => isOrgSelected('All taxonomies')}
      onClick={() => setSelectedOrgFilter('All taxonomies')}
    >
      { isOrgSelected('All taxonomies')
        ? intl.formatMessage(messages.orgInputSelectDefaultValue)
        : intl.formatMessage(messages.orgAllValue)}
    </MenuItem>
    <MenuItem
      key="unassigned-taxonomies"
      className="x-small"
      iconAfter={() => isOrgSelected('Unassigned')}
      onClick={() => setSelectedOrgFilter('Unassigned')}
    >
      { intl.formatMessage(messages.orgUnassignedValue) }
    </MenuItem>
    {isOrganizationListLoaded && organizationListData &&     
      organizationListData.map(org => (
        <MenuItem
          key={`${org}-taxonomies`}
          className="x-small"
          iconAfter={() => isOrgSelected(org)}
          onClick={() => setSelectedOrgFilter(org)}
        >
          {org}
        </MenuItem>,
      )
    )}
  </>;
```

Reply:
I tried following this approach, however the extra `<>` or `div` that holds the `MenuItem`s seems to break the functionality of the `SelectMenu` component. When I click on an item, the menu doesn't 'select' it, it just gets checked in the dropdown and it stays open. So I kept it as is, and just replaced the `map` with `forEach` like you mentioned in the previous comment.